### ISSUE TITLE
How to Define Optimizer for it , since parameters() is not included 

### ISSUE THREAD

**Issue by ankit-rl4** [awsaf49/TransUNet-tf#2]
Hi, I am trying to define a optimizer using 

`optimizer = torch.optim.Adam(model.parameters(), lr=1e-4)
`

but this is no parameters() function , is there a alternate way to do it ?

**Comment by awsaf49** [comment]
Hi, This repo is TF build so `model.parameters()` won't work like torch. You can do something like this in TensorFlow,
```py
optimizer = tf.keras.optimizers.Adam(learning_rate=0.01)
optimizer.apply_gradients(zip(grads, model.trainable_variables))
```

**Comment by ankit-rl4** [comment]
Hi, Thanks a lot. It is Fixed if you don't mind me asking. I am trying to train this model on [breakhis ](https://web.inf.ufpr.br/vri/databases/breast-cancer-histopathological-database-breakhis/)breast cancer dataset where input is 3 channel and  masks is 1 channel . what is the expected output dimensions and i/p dimensions of this model ? and i should be passing 1 tensor to input layer so how should i format it to have image and mask

**Comment by awsaf49** [comment]
You can refer to this notebook for usage, https://www.kaggle.com/code/awsaf49/uwmgi-transunet-2-5d-train-tf
